### ISSUE TITLE
Why The <script> String?

### ISSUE THREAD

**Issue by tomatau** [rackt/async-props#4]
Nice library, it will solve many problems!

So, attempting to render a string `"<script>"` block in a react component used for server side rendering is way more ugly than it should be, needing to completely take control of a containing element.

So sure, no problem yeh, we can:

``` js
<script
  dangerouslySetInnerHTML={{
    __html: `window.__ASYNC_PROPS__ = '${JSON.stringify(asyncProps.propsArray)}';`,
  }}
/>
```

But, why have the script string... an object is far more composable, it's a primitive structure. The script string seems to just overcomplicate things?

Why the generated script string?
Also, why should `__ASYNC_PROPS__` contain a serialised JSON instead of just a normal JS object?

**Comment by rexxars** [comment]
+1. I'm having trouble understanding the hydration currently. 

The script tag returned by `loadPropsOnServer` has the following shape:

``` js
<script>__ASYNC_PROPS__ = [
  { /* ... */ }
]</script>
```

Ergo, an actual array is inserted. However, on the client side, it does a `JSON.parse(__ASYNC_PROPS__)`, which basically ends up as `JSON.parse([object Object])`, which obviously fails.

Surely this can't be how it was intended?

**Comment by tomatau** [comment]
@rexxars, the server hydrates to a script tag where the array is actually inside a string... it's very obscure! Note the `'${JSON.stringify(asyncProps.propsArray)}';` in my snippet.

**Comment by rexxars** [comment]
The code you posted does, but the script tag returned by `loadPropsOnServer` doesn't seem to.

``` js
loadPropsOnServer(renderProps, (loadErr, asyncProps, scriptTag) => {
    console.log(scriptTag);
}
```

Results in:

``` js
<script>__ASYNC_PROPS__ = [
  {
    "blogPosts": []
  }
]</script>
```

I had to use your method of manually JSON.stringifying the propsArray to get it to work.

**Comment by tomatau** [comment]
You're right!
https://github.com/rackt/async-props/blob/master/modules/AsyncProps.js#L120-L130
Should definitely drop the JSON.parse

**Comment by rexxars** [comment]
The tests are passing because the beforeEach step sets the `JSON.stringify` return value directly to a variable which ends up as a string , while the actual code that creates the script tag simply generates HTML, so it ends up as an array. Creating a pull request.

**Comment by zackify** [comment]
I agree, this is weird. I just removed the JSON.parse from the hydrate function for now

**Comment by AlastairTaft** [comment]
+1 I just hit this issue, came here to see if a pull request already existed.

**Comment by zackify** [comment]
I'm about to fork this project, I really need this stuff merged :/

**Comment by ryanflorence** [comment]
The reason its a string is so that you don't have to care about that part and so `AsyncProps` can hydrate automatically, no user-land code in the client at all, you simply have to dump that string onto the page on your server.

The stringifying stuff was just a mistake, it happens :) I merged #7

**Comment by zackify** [comment]
@ryanflorence awesome! Thanks
